lib/reverse_words.rb

```diff
@@ -1,6 +1,25 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity:  O(n) where n is the length of the input string. Each character of the string is
+# iterated over once with a constant number of actions, then each word it iterated over 1/2 the word length again.
+# So it comes out to about O(n + 1/2 n), which reduces to O(n)
+# Space complexity: O(1) constant where n is the length of the my_words string. For any length of the string
+# I use the same number of fixed size variables (either size int or size char).
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  return false if !my_words
+  first_pointer = 0
+  while first_pointer < my_words.length
+    first_pointer += 1 while my_words[first_pointer] == " "
+    second_pointer = first_pointer
```

Comment:
I like these variable names. I can clearly visualize two fingers pointing to positions within an array of words.